[Enhancement] 버튼 컴포넌트 추가

HSjjs98 opened the pull request:
# 🚀 풀 리퀘스트 제안

<!-- AI에게 리뷰를 받고 싶지 않다면 🤷‍♀️이모지를 제거해주세요. -->
@coderabbitai: i🤷‍♀️gnore

## 📋 작업 내용

버튼 컴포넌트 2 종류 추가했습니다.
1. FullButton
2. LongButton

## 🔧 변경 사항

- [ ] 📃 README.md
- [ ] 📦 package.json
- [ ] 🔥 파일 삭제
- [ ] 🧹 그 외 ex) .gitignore 등

app.tsx에 버튼 관련 코드 추가해서 run dev 실행하면 만든 버튼들을 보실 수 있습니다

## 📸 스크린샷 (선택 사항)

수정된 화면 또는 기능을 시연할 수 있는 스크린샷을 첨부해 주세요.

## 📄 기타

추가적으로 전달하고 싶은 내용이나 특별한 요구 사항이 있으면 작성해 주세요.








<!-- This is an auto-generated comment: release notes by OSS CodeRabbit -->
### Summary by CodeRabbit

```
### Release Notes

#### New Feature
- 추가: `FullButton`, `LongButton`, `FitButton` 컴포넌트가 추가되어 다양한 스타일의 버튼을 사용할 수 있습니다.
- 변경: `App` 컴포넌트에 새로운 버튼들이 추가되었습니다.

#### Refactor
- 리팩토링: `theme` 파일명을 대문자로 변경하여 일관성을 유지했습니다.
```
<!-- end of auto-generated comment: release notes by OSS CodeRabbit -->

Commit:
feat: 버튼 컴포넌트 추가 (#13)
- FullButton, LongButton 추가

Commit:
rename: theme.ts 폴더명 소문자로 변경 (#13)

Commit:
feat: 절대경로 사용 및 인터페이스 추가 (#13)
- 버튼 컴포넌트에 인터페이스 추가
- App.tsx에서 절대경로 사용

Commit:
feat: 불필요한 코드 제거와 border:0으로 수정 (#13)

devdeun commented:
FullButton과 LongButton 둘 다 `width: 100%`인 버튼처럼 느껴져요!
기존의 FullButton을 FitButton(FitTextButton..?)으로, LongButton을 FullButton으로 바꾸는거 어떠신가요..?

devdeun commented on src/components/common/buttons/FullButton.tsx:
```diff
@@ -0,0 +1,61 @@
+  customStyle?: SerializedStyles;
+}
+
+interface Theme {
+  colors: {
+    primary: string;
+    white: string;
+    lightestGray: string;
+    darkestGray: string;
+  };
+}
```
이거 직접 만들지 말고
`import { Theme } from '@emotion/react';` 로 불러와서 Theme 타입 적용하면 됩니다!
LongButton도 마찬가지!

devdeun commented on src/components/common/buttons/LongButton.tsx:
```diff
@@ -0,0 +1,75 @@
+  children: ReactNode;
+  onClick?: () => void;
+  type?: 'button' | 'submit' | 'reset';
+  color: 'primary' | 'lightestGray' | 'gray' | 'white';
```
colors로 버튼의 색상을 결정하기보다는 primary, secondary, text, disabled 이런식으로 값을 받는게 더 좋을 것 같아요
disabled일때는 클릭 안되게 만들어야되니까..!

Commit:
design: 디자인 수정에 따른 코드 변경 (#13)
- LongButton 비활성 상태일 때의 배경 색깔 변경

Commit:
style: 버튼 컴포넌트 코드 수정 (#13)
- 인터페이스 임포트
- 컬러 대신 스타일타입 지정
- FitButton, FullButton으로 파일명 수정
- 위 수정 사항에 따른 app.tsx 수정

Commit:
feat: customStyle의 기본값을 빈 배열로 설정 (#13)

Commit:
Merge branch 'develop' of https://github.com/Dev-FE-1/Toy_Project_3_team4 into enhancement/button-compnent-13

Commit:
style: 폰트 사이즈 관련 코드 수정 (#13)

Commit:
fix: 폰트 사이즈를 theme 에서 가져오는 코드 수정 (#13)
